eth_pydantic_types/abi.py

```diff
@@ -0,0 +1,105 @@
+"""
+These models are used to match the lowercase type names used by the abi.
+"""
+
+from typing import ClassVar
+
+from pydantic_core.core_schema import CoreSchema, str_schema
+
+from .address import Address
+from .hex import _make_cls as _make_hex_cls
+from .numbers import _make_cls as _make_number_cls
+
+bytes1 = _make_hex_cls(1, bytes, dict_additions=dict(abi_type="bytes1"))
+bytes4 = _make_hex_cls(4, bytes, dict_additions=dict(abi_type="bytes4"))
+bytes8 = _make_hex_cls(8, bytes, dict_additions=dict(abi_type="bytes8"))
+bytes16 = _make_hex_cls(16, bytes, dict_additions=dict(abi_type="bytes16"))
+bytes20 = _make_hex_cls(20, bytes, dict_additions=dict(abi_type="bytes20"))
+bytes32 = _make_hex_cls(32, bytes, dict_additions=dict(abi_type="bytes32"))
+bytes64 = _make_hex_cls(64, bytes, dict_additions=dict(abi_type="bytes64"))
+
+
+class string(str):
```

Comment:
We can just use `string: TypeAlias = str`, it shouldn't have the weird problems as the dynamic types.